[Soom] L1_Step2_getAreas

developersoom opened the pull request:
**STEP2. 여러 함수를 호출하는 함수**

1. 모든 도형의 넓이 값을 얻을 수 있는 getArea 함수 생성
2. 함수를 호출할 때 마다 누적된 함수 호출 횟수를 알려주는 printExecutionSequence 함수 생성

Commit:
W1_L1_Step2

getArea() function created
printExecutionSequence(); function created

crongro commented:
커밋로그... 좀더 잘 해보세요~
이번까지만 봐드립니다 ㅎ

crongro commented on W1_L1_getAreasStep2.js:
```diff
@@ -0,0 +1,77 @@
+const logArr = [];
+
+getArea('rect',3,4);
```
함수를 호출하는 부분이, 가급적 선언부 이후에 있으면 좋겠어요.

developersoom replied:
수정에 반영하겠습니다~

crongro commented on W1_L1_getAreasStep2.js:
```diff
@@ -0,0 +1,77 @@
+    if (shape === 'rect') {
+        logArr.push('rect');
+        let result = measureRectangleArea(val1, val2);
+        return console.log('사각형의 넓이는 '+result);
```
앗 return 이 틀렸는데 뭐가 이상할까요?

developersoom replied:
console.log()의 반환값은 undefined 입니다. ㅠ.ㅠ

crongro commented on W1_L1_getAreasStep2.js:
```diff
@@ -0,0 +1,77 @@
+        logArr.push('trapezoid');
+        let result = measureTrapezoidArea(val1, val2, val3);
+        return console.log('사다리꼴의 넓이는 '+result);
+    } else if (shape === 'circle' && !val2 && !val3){
```
circle 이면 val2, val3를 체크하는게 불필요하지 않을까요?

developersoom replied:
circles 구하는 것도 있으니 !val3만 빼면 될 것 같습니다~

crongro commented on W1_L1_getAreasStep2.js:
```diff
@@ -0,0 +1,77 @@
+        let result = measureCircleArea(val1);
+        return console.log('원의 넓이는 '+result);
+    } else if (shape === 'circle' && !val3){
+        logArr.push('circle');
```
사실, 넓이를 계산하는 부분이 끝나고 난 후 (즉 메인로직이 잘 성공한 뒤에)
로그를 기록하는 게 정확하죠. 
지금경우는 push로 로깅을 하고, 연산을 하는데 연산이 실패한 경우가 있을수도 있죠. 실패한 경우에도 push로 로깅을 하는 것이라면 모를까..

developersoom replied:
수정해보겠습니다.

crongro commented on W1_L1_getAreasStep2.js:
```diff
@@ -0,0 +1,77 @@
+    for (i = min; i < max + 1; i++) {
+        areas.push(p * Math.pow(i, 2));
+    }
+    let circleAreas = areas.toString();
```
toString을 하면 어떻게 되요?

developersoom replied:
배열 areas의 값들이 문자열로 변환됩니다.

developersoom replied:
> 배열 areas의 값들이 문자열로 변환됩니다.

앗 그럼 다시 숫자로 변환해줘야할까요...?

crongro commented on W1_L1_getAreasStep2.js:
```diff
@@ -0,0 +1,77 @@
+}
+
+function printExecutionSequence(){
+    for (i = 0; i < logArr.length; i++){
```
배웠던 forEach를 쓸 수도 있고요.
배열이나 문자열을 탐색할때는 for-of를 쓸수도 있어요.

developersoom replied:
수정에 반영해보겠습니다~

crongro reviewed (changes requested):
동작이 잘 되도록 구현하셨어요.

리뷰 남겨드리니 조금 수정해보세요~

developersoom committed:
[Soom] L1_Step2_getAreas_edited  #36

1) 함수 호출 부분을 선언부 이후로 수정
2) return 값에 console.log 수정
3) 불필요한 값 삭제(circle val3)
4) push 로그 메인 로그 이후로 이동
5) for 문을 forEach()로 수정

developersoom commented:
[Soom] Step 2 수정했습니다~

crongro commented on W1_L1_getAreasStep2.js:
```diff
@@ -1,28 +1,26 @@
-        logArr.push('rect');
         let result = measureRectangleArea(val1, val2);
-        return console.log('사각형의 넓이는 '+result);
+        logArr.push('rect');
```
그러고보니, logArr.push와 console.log()를 출력하는 두 줄은 
여기 있어도 되지만, 각 함수안에(measureTrapezoidArea등) 있어도 되는거네요.
함수로 옮겨지면 이쪽 분기분이 좀더 간결해질거 같고요. 대신 각 함수가 좀더 지저분(?)해지겠죠? 장단점이 있는거 같습니다.

crongro reviewed (approved):
잘 하셨어요!

const와 let의 차이를 알고, 먼저 const를 써보는 것을 권장합닙다. 안되면 let사용. 물론 var도 공부..

소수점자리를 줄여서 원하는 만큼만 출력하는 방법을 찾아보세요. 

이번렉쳐 수고하셨어요! 
머지합니다.
